spirv_hlsl.cpp

```diff
@@ -4728,21 +4728,33 @@ void CompilerHLSL::emit_subgroup_op(const Instruction &i)
 	case OpGroupNonUniformBallotBitCount:
 	{
 		auto operation = static_cast<GroupOperation>(ops[3]);
+		bool forward = should_forward(ops[4]);
 		if (operation == GroupOperationReduce)
 		{
-			bool forward = should_forward(ops[4]);
 			auto left = join("countbits(", to_enclosed_expression(ops[4]), ".x) + countbits(",
 			                 to_enclosed_expression(ops[4]), ".y)");
 			auto right = join("countbits(", to_enclosed_expression(ops[4]), ".z) + countbits(",
 			                  to_enclosed_expression(ops[4]), ".w)");
 			emit_op(result_type, id, join(left, " + ", right), forward);
 			inherit_expression_dependencies(id, ops[4]);
 		}
-		else if (operation == GroupOperationInclusiveScan)
-			SPIRV_CROSS_THROW("Cannot trivially implement BallotBitCount Inclusive Scan in HLSL.");
-		else if (operation == GroupOperationExclusiveScan)
-			SPIRV_CROSS_THROW("Cannot trivially implement BallotBitCount Exclusive Scan in HLSL.");
-		else
+		else if (operation == GroupOperationInclusiveScan) {
+			auto expr = join("countbits(", to_enclosed_expression(ops[4]), " & gl_SubgroupLeMask)");
+			emit_op(result_type, id, expr, forward);
+			if (!active_input_builtins.get(BuiltInSubgroupLeMask))
+			{
+				active_input_builtins.set(BuiltInSubgroupLeMask);
+				force_recompile_guarantee_forward_progress();
+			}
+		} else if (operation == GroupOperationExclusiveScan) {
+			auto expr = join("countbits(", to_enclosed_expression(ops[4]), " & gl_SubgroupLtMask)");
+			emit_op(result_type, id, expr, forward);
+			if (!active_input_builtins.get(BuiltInSubgroupLtMask))
+			{
+				active_input_builtins.set(BuiltInSubgroupLtMask);
+				force_recompile_guarantee_forward_progress();
+			}
+		} else
```

Comment:
From a quick look in that file upstream seems to favor Allman style:
```suggestion
		else if (operation == GroupOperationInclusiveScan)
		{
			auto expr = join("countbits(", to_enclosed_expression(ops[4]), " & gl_SubgroupLeMask)");
			emit_op(result_type, id, expr, forward);
			if (!active_input_builtins.get(BuiltInSubgroupLeMask))
			{
				active_input_builtins.set(BuiltInSubgroupLeMask);
				force_recompile_guarantee_forward_progress();
			}
		}
		else if (operation == GroupOperationExclusiveScan)
		{
			auto expr = join("countbits(", to_enclosed_expression(ops[4]), " & gl_SubgroupLtMask)");
			emit_op(result_type, id, expr, forward);
			if (!active_input_builtins.get(BuiltInSubgroupLtMask))
			{
				active_input_builtins.set(BuiltInSubgroupLtMask);
				force_recompile_guarantee_forward_progress();
			}
		}
		else
```

Reply:
Done!